src/MiniCover/HitServices/HitService.cs

```diff
@@ -68,14 +33,29 @@ public MethodContext(ConcurrentDictionary<int, Hit> hitInstructions)
 
             public void HitInstruction(int id)
             {
-                var hitInstruction = hitInstructions.GetOrAdd(id, i => new Hit(i));
-                hitInstruction.HitedBy(testMethod);
+                testMethod.HasHit(id);
+
             }
 
             public void Exit()
             {
                 if (clearTestMethodCache)
+                {
+                    this.Save();
                     TestMethodCache.Value = null;
+                }
+            }
+
+            private void Save()
+            {
+                using (var fileStream = File.Open(this.filePath, FileMode.Append, FileAccess.Write, FileShare.None))
```

Comment:
I think we would need a lock on save now, because we might have multiple instances of MethodContext Saving data on the same moment when you run parallel code.